trl/trainer/rloo_trainer.py

```diff
@@ -95,6 +95,10 @@ def __init__(
                 "same as `policy`, you must mass a copy of it, or `None` if you use peft."
             )
 
+        # Define the collator if not provided
+        if data_collator is None:
+            data_collator = DataCollatorWithPadding(self.processing_class)
```

Comment:
move this further down as `self.processing_class` is not defined here... or perhaps do it all inline?
```
self.data_collator =  DataCollatorWithPadding(self.processing_class) if data_collator is None else data_collator
```

Reply:
Hi, of course you are right! Sorry for the silly mistake. 

Hopefully this is fine right now.